packages/event-storage-adapter-postgres/src/adapter.ts

```diff
@@ -0,0 +1,484 @@
+/* eslint-disable complexity */
+
+/* eslint-disable max-lines */
+import type { SerializableParameter } from 'postgres';
+import postgres from 'postgres';
+
+import { GroupedEvent } from '@castore/core';
+import type {
+  EventDetail,
+  EventsQueryOptions,
+  EventStorageAdapter,
+  EventStoreContext,
+  ListAggregateIdsOptions,
+  ListAggregateIdsOutput,
+  OptionalTimestamp,
+  PushEventOptions,
+} from '@castore/core';
+
+import { PostgresEventAlreadyExistsError } from './error';
+
+const assertIsSerializableParameter: (
+  value: unknown,
+) => asserts value is SerializableParameter = value => {
+  if (!value) {
+    throw new Error(`Payload is required`);
+  }
+
+  if (typeof value !== 'object' || Array.isArray(value)) {
+    throw new Error(`Payload must be a plain object`);
+  }
+};
+
+export type ParsedPageToken = {
+  limit?: number;
+  initialEventAfter?: string | undefined;
+  initialEventBefore?: string | undefined;
+  reverse?: boolean | undefined;
+  lastEvaluatedKey?:
+    | {
+        aggregateId: string;
+        initialEventTimestamp: string;
+      }
+    | undefined;
+};
+
+export class PostgresEventStorageAdapter implements EventStorageAdapter {
+  private _sql: postgres.Sql<{ bigint: bigint }>;
+
+  constructor(
+    payload: { connectionString: string } = {
```

Comment:
This will be simpler to update in the future:

```ts
constructor( { connectionString = 'postgresql...' }: { connectionString?: string } = {}) {
  ...
}
```